docs/specs/instrumentation.md

```diff
@@ -8,34 +8,143 @@ interpreted as described in [RFC 2119].
+
+* Vector process batches of events ([RFC 9480]) and telemtry SHOULD emit for
+  the entire batch, not each individual event.
+
+### Events
+
+Instrumentation SHOULD be event-driven ([RFC 2064]), where individual events
+serve as the vehicle for internal telemtry, driving the emission of metrics
+and logs. This organizes Vector's telemetry, making it easier to manage and 
+catalogue. On rare occassions, metrics and logs can emit directly but MUST be
+reserved for ocassions where it is impossible to emit Vector's events. For
+example, emitting metrics in a library that cannot import Vector's events.
+
+#### Error
+
+An `<Name>Error` event MUST be emitted when an error occurs. Errors that are
+retriable and possible recoverable MUST be distinguished from errors that are
+not:
+
+* Properties
+  * `retriable` - If the operation that produced the error is retriable. This
+    indicates that the error operation might recover and determines if this
+    error should be a warning or an error when emitting logs and metrics. For
+    example, a failed HTTP request that will be retried.
+  * `error_code` - An error code for the failure, if applicable.
+    * SHOULD only be specified if it adds additional information beyond
+      `error_type`.
+    * The values for `error_code` for a given error event MUST be a bounded set
+      with relatively low cardinality because it will be used as a metric tag.
+      Examples would be syscall error code. Examples of values that should not
+      be used are raw error messages from `serde` as these are highly variable
+      depending on the input. Instead, these errors should be converted to an
+      error code like `invalid_json`.
+  * `error_type` - The type of error condition. MUST be one of the types listed
+    in the `error_type` enum list in the cue docs.
+  * `stage` - The stage at which the error occurred. MUST be one of `receiving`,
+    `processing`, or `sending`.
+  * If any of the above properties are implicit to the specific error
+    type, they MAY be omitted from being represented explicitly in the
+    event fields. However, they MUST still be included in the emitted
+    logs and metrics, as specified below, as if they were present.
+* Metrics
+  * MUST include the defined properties as tags.
+  * If `retriable` is `true`, MUST increment `<namespace>_warnings_total` metric.
+  * If `retriable` is `false`, MUST increment `<namespace>_errors_total` metric.
+* Logs
+  * MUST log a descriptive, user friendly error message that sufficiently
+    describes the error.
+  * MUST include the defined properties as key-value pairs, except `message`.
+  * If `retriable` is `true`, MUST log a message at the `warning` level.
+  * If `retriable` is `false`, MUST log a message at the `error` level.
+  * SHOULD be rate limited to 10 seconds.
+* Events
+  * MUST emit an [`EventsDropped`] event if the error results in dropping events,
+    or the error itself MUST meeting the `EventsDropped` requirements.
```

Comment:
Shifted the `*EventsDropped` event requirements below, noting that the error itself can meet the requirements or emit another event.